build/ci/community/create_demo_pr.sh

```diff
@@ -1,38 +1,57 @@
-#!/bin/bash -xe
+#!/bin/bash -xel
 set +o pipefail
 
-yq -y --arg operator_image $operator_image_for_test '.spec.install.spec.deployments[0].spec.template.spec.containers[0].image |= $operator_image' $csv_file > temp.yaml
-cat temp.yaml > $csv_file
-yq -y --arg operator_image $operator_image_for_test '.metadata.annotations.containerImage |= $operator_image' $csv_file > temp.yaml
-cat temp.yaml > $csv_file
-yq -y --arg operator_image $operator_image_for_test '.spec.relatedImages[0].image |= $operator_image' $csv_file > temp.yaml
-cat temp.yaml > $csv_file
-
-echo $github_token > github_token.txt
-gh auth login --with-token < github_token.txt
-gh repo fork operator-framework/community-operators --clone community-operators-fork
-cd community-operators-fork
-git remote set-url origin https://csiblock:$github_token@github.com/csiblock/community-operators.git
-git fetch upstream
-git rebase upstream/master
-git push origin master --force
+edit_operator_image_in_csv_yaml_file (){
+  cd $(dirname $csv_file)
+  chmod 547 $(basename $csv_file) 
+  declare -a operator_image_fields=(
+      ".spec.install.spec.deployments[0].spec.template.spec.containers[0].image"
+      ".metadata.annotations.containerImage"
+      ".spec.relatedImages[0].image"
+  )
+  for image_field in "${operator_image_fields[@]}"
+  do
+      yq eval "$image_field |= env(operator_image_for_test)" $(basename $csv_file) -i
+  done
+cd -
+}
 
 create_demo_pr(){
   community_operators_branch=$1
   dest_path=$2
-  cluster_kind=$3  
-  export repo_pr=`gh pr list --repo $github_csiblock_community_operators_repository | grep $community_operators_branch`
-  if [[ "$repo_pr" == *"$community_operators_branch"* ]]; then
-    gh pr close $community_operators_branch --delete-branch --repo $github_csiblock_community_operators_repository
+  cluster_kind=$3
+  forked_repository=$4
+  cd $forked_repository-fork
+  repo_pr="gh pr list --repo $forked_repository | grep $community_operators_branch"
+  if [[ "`eval $repo_pr`" == *"$community_operators_branch"* ]]; then
+    gh pr close $community_operators_branch --delete-branch --repo $forked_repository
   fi
-  git checkout master
+  git checkout main
   git checkout -b $community_operators_branch
   yes | cp -r $repository_path/deploy/olm-catalog/ibm-block-csi-operator-community/ $dest_path
   git add .
   git commit --signoff -m "build number $github_build_number $cluster_kind"
   git push origin $community_operators_branch
-  gh pr create --title "IBM Block CSI update $cluster_kind" --repo $github_csiblock_community_operators_repository --base master --head $community_operators_branch --body "pr check"
+  gh pr create --title "IBM Block CSI update $cluster_kind" --repo $forked_repository --base main --head $community_operators_branch --body "pr check"
+  cd -
+}
+
+update_community_operators_fork (){
```

Comment:
```suggestion
update_forked_community_operators_repo (){
```

Reply:
@matancarmeli7 ?

Reply:
> you have a fork of the Spoon-Knife repository
from my understanding of english, `a fork of the X repo` == `repo X fork`

Reply:
ok